sound/soc/sof/ipc3.c

```diff
@@ -632,7 +936,7 @@ static void sof_ipc3_rx_msg(struct snd_sof_dev *sdev)
 	case SOF_IPC_FW_READY:
 		/* check for FW boot completion */
 		if (sdev->fw_state == SOF_FW_BOOT_IN_PROGRESS) {
-			err = sof_ops(sdev)->fw_ready(sdev, cmd);
+			err = ipc3_fw_ready(sdev, cmd);
```

Comment:
this change does two things
a) make fw_ready IPC3 specific - ok
b) remove the platform-dependent implementation. that's the part I am not sure of, what makes you think it is ok or safe to remove the platform abstraction? This is after all related to the firmware boot process which is highly platform-specific, no?

Reply:
> this change does two things
> a) make fw_ready IPC3 specific - ok

It makes the fw_ready to be IPC version specific, yes we have only IPC3 atm.

> b) remove the platform-dependent implementation. that's the part I am not sure of, what makes you think it is ok or safe to remove the platform abstraction? This is after all related to the firmware boot process which is highly platform-specific, no?

This message suppose to let the host know that the firmware is up and running. Basically the firmware sends an indication which we receive via normal rx_msg path.
The platform code passes this up to the IPC layer, if it needs to do something specific other than the reception and passing forward, then it can do.

But from this point onward it is up to the IPC (and firmware) implementation to handle what to do, in my understanding.

Reply:
ok, makes sense then.